### PR TITLE
Update comments with I-Ds to their RFCs

### PR DESCRIPTION
**Pull request by ksmurchison**
This literally just updates one document and a bunch of source code comments

### CODE REVIEW

File: sieve/bytecode.h [hunk 1]
```diff
@@ -132,7 +132,7 @@ typedef union
  * version 0x13 scripts forced an upgrade for broken bytecode generated by a buggy commit
  * version 0x14 scripts implemented Environment (RFC 5183)
  * version 0x15 scripts implemented Redirect-DeliverBy, Redirect-DSN (RFC 6009)
- * version 0x16 scripts implemented draft-gondwana-sieve-mailboxid-01
+ * version 0x16 scripts implemented MailboxId 9042
```
Comment (dilyanpalauzov) [inline sieve/bytecode.h:135]: The word `RFC` is missing.

**Comment by dilyanpalauzov** [comment]
How about changing in annotate.c draft-ietf-specialuse-important with RFC 8457?

jmap_contacts.c contains twice draft-ietf-calext-vcard-jscontact-extensions, but the newest version is called RFC 9554. download/installation/http/jmap and rfc-support.rst do not say anything about draft-ietf-calext-vcard-jscontact-extensions/RFC 9554.

**Comment by dilyanpalauzov** [comment]
If JMAP Push Subscription is implemented, as there as an option `jmappushsubscriptionfolder`, I suggest to remove in installation/http/jmap.rst `, except for PushSubscription` and add JMAP Push Subscription to the right release notes.  If it is not implemented, I propose adding in rfc-support.rst a `.. NOTE:` after RFC 8620.

**Review by elliefm — APPROVED** [review]

**Comment by dilyanpalauzov** [comment]
Is replacing in changes/next/imap-partial draft-ietf-extra-imap-inprogress with RFC 9585 appropriate?

In imap/http_dav.c change draft-ietf-calext-caldav-attachments with RFC 8607.

Does 3.10 add support for draft-ietf-extra-imap-uidonly, as stated in the release notes for .10.0-rc1, .10.0-rc2, 3.10.0-beta1, 3.10.0-beta and 3.10, or it implements also RFC 9586?

**Comment by ksmurchison** [comment]
> Does 3.10 add support for draft-ietf-extra-imap-uidonly, as stated in the release notes for 3.10.0-rc1, 3.10.0-rc2, 3.10.0-beta1, 3.10.0-beta and 3.10, or it implements also RFC 9586?

It implemented whatever draft version was available at the time.  I don't recall if it changed significantly between that draft and being published as RFC 9586.  I don't think we want to change release notes, but I'll leave that up to @elliefm

**Pull request merged**